modin/backends/base/query_compiler.py

```diff
@@ -213,88 +228,102 @@ def reset_index(self, **kwargs):
     # we will implement a Distributed Series, and this will be returned
     # instead.
 
+    @abc.abstractmethod
     def count(self, **kwargs):
         """Counts the number of non-NaN objects for each column or row.
 
         Return:
             Pandas series containing counts of non-NaN objects from each column or row.
         """
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def max(self, **kwargs):
         """Returns the maximum value for each column or row.
 
         Return:
             Pandas series with the maximum values from each column or row.
         """
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def mean(self, **kwargs):
         """Returns the mean for each numerical column or row.
 
         Return:
             Pandas series containing the mean from each numerical column or row.
         """
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def min(self, **kwargs):
         """Returns the minimum from each column or row.
 
         Return:
             Pandas series with the minimum value from each column or row.
         """
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def prod(self, **kwargs):
         """Returns the product of each numerical column or row.
 
         Return:
             Pandas series with the product of each numerical column or row.
         """
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def sum(self, **kwargs):
         """Returns the sum of each numerical column or row.
 
         Return:
             Pandas series with the sum of each numerical column or row.
         """
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
     # END Abstract full Reduce operations
 
     # Abstract map partitions operations
     # These operations are operations that apply a function to every partition.
+    @abc.abstractmethod
     def abs(self):
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def applymap(self, func):
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def isin(self, **kwargs):
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def isna(self):
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
     def isnull(self):
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        return self.isna()
 
+    @abc.abstractmethod
     def negative(self, **kwargs):
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
+    @abc.abstractmethod
     def notna(self):
-        raise NotImplementedError(NOT_IMPLEMENTED_MESSAGE)
+        pass
 
     def notnull(self):
```

Comment:
`notnull` can be deleted we want to only expose one way to do `notna`.

`notnull = notna` in the API layer: https://github.com/modin-project/modin/blob/9e21b589d0baaee3de4f31fe2171f3dfaa9363a7/modin/pandas/base.py#L1756